src/decideid_assets/src/components/auth/new_profile/NewProfile.tsx

```diff
@@ -1,18 +1,34 @@
 import React, { useState } from "react";
 import { useProfile } from "../../../utils";
+import { useNavigate } from "react-router-dom";
+import { useEffect } from "react";
 
 export default function NewProfile() {
+  const navigate = useNavigate();
+  const { createProfile, profile, isLoading } = useProfile();
   const [firstName, setFirstName] = useState('');
   const [lastName, setLastName] = useState('');
   const [email, setEmail] = useState('');
-  const { createProfile } = useProfile()
 
   const handleCreateProfile = async () => {
     const newProfile = { firstName, lastName, email };
-    await createProfile(newProfile);
+    createProfile(newProfile);
   };
 
-  return (
+  useEffect(() => {
+    if (!isLoading) {
+      if (profile?.message == null && profile?.email != null) {
+        navigate('/app');
+      }
+    }
+  }, [isLoading, profile, navigate]);
+
+  return <>
+    { profile?.message ? (
```

Comment:
what is `Message` related to `Profile` ?

Reply:
Thanks. You are correct. Let me modify this.